fetch-engine.sh

```diff
@@ -20,29 +20,39 @@ fi
 if [ "${AUTOMATIC_ENGINE_MANAGEMENT}" = "True" ]; then
 	echo "OpenRA engine version ${ENGINE_VERSION} is required."
 
-	if [ -d "${ENGINE_DIRECTORY}" ]; then
-		if [ "${CURRENT_ENGINE_VERSION}" != "" ]; then
-			echo "Deleting engine version ${CURRENT_ENGINE_VERSION}."
-		else
-			echo "Deleting existing engine (unknown version)."
+	if command -v git >/dev/null 2>&1; then
+		if [ ! -d "${ENGINE_DIRECTORY}/.git" ]; then
+			rm -rf "${ENGINE_DIRECTORY}"
+			git clone "${AUTOMATIC_ENGINE_SOURCE_GIT}" "${ENGINE_DIRECTORY}"
 		fi
 
-		rm -rf "${ENGINE_DIRECTORY}"
-	fi
+		git --git-dir="${ENGINE_DIRECTORY}/.git" --work-tree="${ENGINE_DIRECTORY}" fetch --all
+		git --git-dir="${ENGINE_DIRECTORY}/.git" --work-tree="${ENGINE_DIRECTORY}" checkout "${ENGINE_VERSION}"
```

Comment:
Users must not edit the engine dir when automatic management is enabled.

https://github.com/OpenRA/OpenRAModSDK/blob/6cbff13bfb617d08264c7bf6dac50811c0e97075/mod.config#L75-L77

Advanced users who want to be able to edit their engine should disable automatic management through their user.config.